crates/anvil/src/eth/backend/env.rs

```diff
@@ -0,0 +1,57 @@
+use alloy_evm::EvmEnv;
+use foundry_evm::EnvMut;
+use foundry_evm_core::AsEnvMut;
+use op_revm::{transaction::deposit::DepositTransactionParts, OpTransaction};
+use revm::{
+    context::{BlockEnv, CfgEnv, TxEnv},
+    primitives::hardfork::SpecId,
+};
+
+/// Helper container type for [`EvmEnv`] and [`OpTransaction<TxEnd>`].
+#[derive(Clone, Debug, Default)]
+pub struct Env {
+    pub evm_env: EvmEnv,
+    pub tx: OpTransaction<TxEnv>,
+    pub is_optimism: bool,
```

Comment:
improvement would be to get rid of the is_optimism flag all together but don't see a good way currently to do so